### PR TITLE
Check function parameters before executing the function and script.

### PR DESCRIPTION
**Pull request by steelgeek091**
According to issue https://github.com/rooch-network/rooch/issues/202.

When MoveOS executes functions and scripts, the verifier checks the parameters and return values of the functions before execution, in the same way as it checks the entry function during compile time.

### CODE REVIEW

File: moveos/moveos/src/moveos.rs [hunk 1]
```diff
@@ -190,6 +193,11 @@ impl MoveOS {
                 let args = session
                     .resolve_args(&tx_context, loaded_function, function.args)
                     .map_err(|e| e.finish(Location::Undefined))?;
+
+                let func =
+                    session.load_function(&function.function_id, function.ty_args.as_slice())?;
+                moveos_verifier::verifier::verify_entry_function(func, session.runtime_session())?;
+
```
Comment (jolestar) [inline moveos/moveos/src/moveos.rs:200]: `verify_entry_function` should be before the `resolve_args`.

File: moveos/moveos-verifier/src/metadata.rs [hunk 2]
```diff
@@ -313,25 +333,28 @@ impl<'a> ExtendedChecker<'a> {
...
-                | "0x1::tx_context::TxContext"
-        )
-    }
+pub fn is_allowed_input_struct(name: String) -> bool {
+    matches!(
+        name.as_str(),
+        "0x1::string::String"
+            | "0x1::object_id::ObjectID"
+            | "0x1::storage_context::StorageContext"
+            | "0x1::tx_context::TxContext"
```
Comment (jolestar) [inline moveos/moveos-verifier/src/metadata.rs:356]: 0x1::ascii::String also support.

**Review by jolestar — APPROVED** [review]

**Comment by jolestar** [comment]
So the #152 and #202 can be marked as resolved?

File: moveos/moveos/src/moveos.rs (hunk 1 above)
Comment (steelgeek091) [reply moveos/moveos/src/moveos.rs:200]: Got it. I will submit a pull request to modify it later.

File: moveos/moveos-verifier/src/metadata.rs (hunk 2 above)
Comment (steelgeek091) [reply moveos/moveos-verifier/src/metadata.rs:356]: Okay, I will add it in a new pull request.

**Comment by steelgeek091** [comment]
> So the #152 and #202 can be marked as resolved?

Right, these two issues have been resolved.

**Pull request merged**